backend/src/test/java/com/bakdata/conquery/util/NonPersistentNamespacedCentralRegisteredStorage.java

```diff
@@ -0,0 +1,249 @@
+        imports.add(imp);
+        imp.loadExternalInfos(this);
+
+        for (Concept<?> c : getAllConcepts()) {
+            for (Connector con : c.getConnectors()) {
+                if (con.getTable().getId().equals(imp.getTable())) {
+                    con.addImport(imp);
+                }
+            }
+        }
+        centralRegistry.register(imp);
+    }
+
+    @Override
+    public Import getImport(ImportId id) {
+        return imports.get(id);
+    }
+
+    @Override
+    public Collection<Import> getAllImports() {
+        return imports.values();
+    }
+
+    @Override
+    public void updateImport(Import imp) {
+        removeImport(imp.getId());
+        addImport(imp);
+    }
+
+    @Override
+    public void removeImport(ImportId id) {
+        centralRegistry.remove(id);
+        imports.remove(id);
+    }
+
+    @Override
+    public Dataset getDataset() {
+        return dataset;
+    }
+
+    @Override
+    public void updateDataset(Dataset dataset) {
+        centralRegistry.remove(dataset);
+        this.dataset = dataset;
+        centralRegistry.register(dataset);
+    }
+
+    @Override
+    public List<Table> getTables() {
+        return new ArrayList<>(tables.values());
+    }
+
+    @Override
+    public Table getTable(TableId tableId) {
+        return tables.get(tableId);
+    }
+
+    @Override
+    public void addTable(Table table) {
+        tables.add(table);
+        centralRegistry.register(table);
+
+        for (Column c : table.getColumns()) {
+            centralRegistry.register(c);
+        }
+    }
+
+    @Override
+    public void removeTable(TableId table) {
+        for (Column c : tables.get(table).getColumns()) {
+            centralRegistry.remove(c);
+        }
+        centralRegistry.remove(table);
+
+        tables.remove(table);
+    }
+
+    @Override
+    public List<SecondaryIdDescription> getSecondaryIds() {
+        return new ArrayList<>(secondaryIds.values());
+    }
+
+    @Override
+    public SecondaryIdDescription getSecondaryId(SecondaryIdDescriptionId descriptionId) {
+        return secondaryIds.get(descriptionId);
+    }
+
+    @Override
+    public void addSecondaryId(SecondaryIdDescription secondaryIdDescription) {
+        secondaryIds.add(secondaryIdDescription);
+        centralRegistry.register(secondaryIdDescription);
+    }
+
+    @Override
+    public void removeSecondaryId(SecondaryIdDescriptionId secondaryIdDescriptionId) {
+        centralRegistry.remove(secondaryIdDescriptionId);
+        secondaryIds.remove(secondaryIdDescriptionId);
+    }
+
+    @Override
+    public Concept<?> getConcept(ConceptId id) {
+        return concepts.get(id);
+    }
+
+    @Override
+    public boolean hasConcept(ConceptId id) {
+        return concepts.containsKey(id);
+    }
+
+    @Override
+    @SneakyThrows
+    public void updateConcept(Concept<?> concept) {
+        centralRegistry.remove(concept);
+
+        concept.initElements(validator);
```

Comment:
Das ist gefährlich, jetzt haben wir die Logik doppelt. Extrahier das bitte ordentlich so dass das nur einmal pluggable stattfindet.

Reply:
Ja das ist blöde. Eine richtig gute Idee habe ich leider noch nicht

Reply:
Ich habe jetzt extra Methoden in NamespaceStorage implementiert die das Wrapping übernehmen